jcl/src/java.base/share/classes/java/lang/System.java

```diff
@@ -753,16 +753,20 @@ public static void setSecurityManager(final SecurityManager s) {
 			/*[PR 97686] Preload the policy permission */
 			AccessController.doPrivileged(new PrivilegedAction() {
 				public Object run() {
+					ProtectionDomain newDomain = s.getClass().getPDImpl();
+					/*[IF !Java11]*/
 					ProtectionDomain oldDomain = currentSecurity == null ?
```

Comment:
The test should remain for when a security manager is already installed, but being changed. However the test can be modified to pre-initialize when changing the security manager from null to something else.